### PR TITLE
Fix inaccurate comment in SEMS RHEL7 env script

### PR DESCRIPTION
**Pull request by mhoemmen**
@trilinos/framework 

## Description

Fix inaccurate comment.  It's really downstream solvers that need `CUDA_LAUNCH_BLOCKING=1`.  Tpetra runs in SPARC just fine without it.

### CODE REVIEW

File: cmake/std/atdm/sems-rhel7/environment.sh [hunk 1]
```diff
@@ -190,7 +190,6 @@ elif [[ "$ATDM_CONFIG_COMPILER" == "CUDA-9.2" ]] ; then
   export LAPACK_ROOT=/usr/lib64/atlas
   export ATDM_CONFIG_LAPACK_LIBS="-L${LAPACK_ROOT};-llapack"
   export ATDM_CONFIG_BLAS_LIBS="-L${BLAS_ROOT}/lib;-lblas"
-  # Needed for some Tpetra UVM stuff
```
Comment (bartlettroscoe) [inline cmake/std/atdm/sems-rhel7/environment.sh]: That was the case the last time I checked.  I had to add this back specifically to make a bunch of Tpetra tests pass.  When did Tpetra remove this requirement?

If it is true that Tpetra does not need this anymore, then we should just remove this and test to make sure.

Comment (mhoemmen) [reply cmake/std/atdm/sems-rhel7/environment.sh]: @bartlettroscoe Most of _Tpetra_ runs fine without `CUDA_LAUNCH_BLOCKING=1`, but most downstream packages do _not_ run fine.  I merely object to the characterization of Tpetra as the source of all trouble ;-)  @kddevin doesn't need more stress in her life....

Comment (bartlettroscoe) [reply cmake/std/atdm/sems-rhel7/environment.sh]: Then let's change the comment to something like "some Trilinos tests require this to run correctly"

Comment (mhoemmen) [reply cmake/std/atdm/sems-rhel7/environment.sh]: Done; thanks!

**Comment by bartlettroscoe** [comment]
@mhoemmen, looks like a conflict is stopping this from merging

**Comment by prwolfe** [comment]
Conflicts with some input from #5516 - looks like a simple fix.

**Comment by mhoemmen** [comment]
Thanks Ross!

**Pull request merged**